### PR TITLE
code refactor: mount in virt-handler for hotplugged volumes

### PR DESCRIPTION
**Pull request by alicefr**
**What this PR does / why we need it**:
Refactor some functions in `pkg/virt-handler/hotplug-disk/mount.go` and tries to simplify the flow avoiding repetitions

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by alicefr** [comment]
@awels @ShellyKa13 could you please have a look when you have time :)

**Review by ShellyKa13 — COMMENTED** [review]
nice refactoring!
/lgtm

**Review by xpivarc — COMMENTED** [review]
Looks good I have one question below.

**Comment by alicefr** [comment]
Reedit the commit messaged as suggested

File: pkg/virt-handler/hotplug-disk/mount.go [hunk 1]
```diff
@@ -353,6 +339,9 @@ func (m *volumeMounter) MountFromPod(vmi *v1.VirtualMachineInstance, sourceUID t
 		if volumeStatus.MemoryDumpVolume != nil {
 			mountDirectory = true
 		}
+		if sourceUID == types.UID("") {
```
[the remark this answers is not in the document]
Comment (xpivarc) [reply pkg/virt-handler/hotplug-disk/mount.go:342]: Yes, this case is alright but general `Mount` is taking the UID from [volumeStatus](https://github.com/kubevirt/kubevirt/blob/28630a3f91434203f2eac3f4c5bce217e6a47833/pkg/virt-handler/hotplug-disk/mount.go#L334) which suggest to me there could be multiple... I need to look at our current implementation as I know we changed a few things. 

@mhenriks @awels You will probably know right away.

[the remark this answers is not in the document]
Comment (alicefr) [reply pkg/virt-handler/hotplug-disk/mount.go:342]: Oh yes, I see. Alexander can correct me, but I think we changed from an attachPod per hotplugged volume to an attachPod per namespace.

[the remark this answers is not in the document]
Comment (ShellyKa13) [reply pkg/virt-handler/hotplug-disk/mount.go:342]: I can attest to that :)

[the remark this answers is not in the document]
Comment (awels) [reply pkg/virt-handler/hotplug-disk/mount.go:342]: Its a single attachmentPod per VM regardless of how many volumes are hotplugged in the VM

[the remark this answers is not in the document]
Comment (xpivarc) [reply pkg/virt-handler/hotplug-disk/mount.go:342]: Thanks @ShellyKa13 @awels

**Comment by xpivarc** [comment]
/approve
/lgtm
Thanks!

**Pull request merged**